lib/ansible/modules/cloud/amazon/aws_glacier_vault.py

```diff
@@ -0,0 +1,227 @@
+extends_documentation_fragment:
+    - aws
+    - ec2
+'''
+
+RETURN = '''
+vault:
+    description: the vault object
+    returned: always
+    type: complex
+    contains:
+        CreationDate:
+            description: The vault creation date.
+            returned: always
+            type: string
+            sample: "2018-02-13T22:10:28.590Z"
+        NumberOfArchives:
+            description: Number of archives currently in the vault.
+            returned: always
+            type: int
+            sample: 0
+        SizeInBytes:
+            description: Current vault size.
+            returned: always
+            type: int
+            sample: 0
+        VaultARN:
+            description: The vault AWS ARN.
+            returned: always
+            type: string
+            sample: "arn:aws:glacier:eu-west-3:123457788994:vaults/test-vault"
+        VaultName:
+            description: The vault name.
+            returned: always
+            type: string
+            sample: "test-vault"
+'''
+
+EXAMPLES = '''
+# Note: These examples do not set authentication details, see the AWS Guide for details.
+
+# Create a simple glacier vault
+- aws_glacier_vault:
+    name: myglaciervault
+
+# Create a simple glacier vault in eu-west-3 region
+- aws_glacier_vault:
+    name: myglaciervault
+    region: eu-west-3
+
+# Remove an glacier vault
+- aws_glacier_vault:
+    name: myglaciervault
+    state: absent
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import (ec2_argument_spec, boto3_conn, HAS_BOTO3, get_aws_connection_info,
+                                      compare_aws_tags)
+
+try:
+    import botocore
+    from botocore.exceptions import ClientError
+except ImportError:
+    pass  # caught by imported HAS_BOTO3
+
+
+class GlacierVaultManager(object):
+    def __init__(self, connection, module):
+        self.connection = connection
+        self.module = module
+        self.name = self.module.params.get('name')
+
+    def list(self):
+        vaults = self.connection.list_vaults()
+        if "VaultList" not in vaults:
+            self.module.fail_json(msg="Invalid response from Glacier: no VaultList found in response")
+
+        return vaults["VaultList"]
+
+    def exists(self, name):
+        for v in self.list():
+            if "VaultName" not in v:
+                self.module.fail_json(
+                    msg="Invalid response from Glacier: no VaultName found in vault object {0}".format(v))
+
+            if v["VaultName"] == name:
+                return v
+
+        return None
+
+    def create_or_update(self):
+        vault = self.exists(self.name)
+        changed = False
+        if vault is None:
+            vault = self.connection.create_vault(vaultName=self.name)
+            changed = True
+
+        changed |= self.manage_tags()
+
+        # Vault exists, no change to do
+        self.module.exit_json(vault=vault, changed=changed)
```

Comment:
create_vault returns a dict with `location` and `ResponseMetadata`.

You'll probably need to call describe_vault to return the expected parameters (and please use camel_dict_to_snake_dict, and don't forget to also call list_tags_for_vault, convert the tags using boto3_tag_list_to_ansible_dict and then add those to the results too)

Edit: no need to call boto3_tag_list_to_ansible_dict as tags are actually in the right format for once. This service must be unique!